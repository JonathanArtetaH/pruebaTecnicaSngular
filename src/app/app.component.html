<div class="container">
  <div class="row mt-5">
    <div class="col">
      <app-formulario (nuevoNumeroEvent)="cambioNumeroEvent($event)"></app-formulario>
    </div>
    <div class="col">
      <app-mostrar-resultado [resultado]="resultado"></app-mostrar-resultado>
    </div>
    <app-operaciones [numeroInput]="numero"></app-operaciones>
  </div>
</div>