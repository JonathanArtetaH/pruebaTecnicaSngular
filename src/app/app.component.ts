import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  title = 'pruebaTecnicaSngular';
  numero: number  = 0;
  resultado: number = 0; 

  cambioNumeroEvent = (event: number) => this.numero = event;


  obtenerResultado (event: number){ 
    this.resultado = (!isNaN(event))
    ? event
    : 0 
  }
  

}
