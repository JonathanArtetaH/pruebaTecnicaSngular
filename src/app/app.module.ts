import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { ComponentsModule } from './components/components.module';
import { FormsModule } from '@angular/forms';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    ComponentsModule,
    FormsModule
  ], 
  bootstrap: [AppComponent], 
})
export class AppModule { }
