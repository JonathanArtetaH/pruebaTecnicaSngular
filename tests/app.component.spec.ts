import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';   
import { AppComponent } from '../src/app/app.component';

describe('AppComponent', () => {

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });

  
  test('Creacion del componente', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const component = fixture.componentInstance;
    expect(component).toBeTruthy();
  });

  test('Debe coincidir con el snapshot', () => {
    const fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
    const compiled = fixture.nativeElement as HTMLElement;
    expect( compiled ).toMatchSnapshot();
  })

});
