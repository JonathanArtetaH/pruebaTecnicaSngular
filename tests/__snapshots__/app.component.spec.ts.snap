// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Debe coincidir con el snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="container"
  >
    <div
      class="row mt-5"
    >
      <div
        class="col"
      >
        <app-formulario />
      </div>
      <div
        class="col-auto"
      >
        <app-mostrar-resultado />
      </div>
      <app-operaciones />
    </div>
  </div>
</div>
`;
